pandas/core/common.py

```diff
@@ -231,6 +235,11 @@ def asarray_tuplesafe(values, dtype=None):
     if isinstance(values, list) and dtype in [np.object_, object]:
         return construct_1d_object_array_from_listlike(values)
 
+    if isinstance(values, list) and (
+        is_datetime64_any_dtype(dtype) or is_period_dtype(dtype)
+    ):
+        return array(values, dtype)
```

Comment:
does timedelta64 ever go through here?  could just check needs_i8_conversion(dtype)

What about other EA dtypes?  isnt this kind of "anything that we dont want to pass to np.asarray below"?

Reply:
@jbrockmendel 
Okay. I change code